wolff/optimization/curvature_finite_difference_gradient_model.py

```diff
@@ -0,0 +1,152 @@
+from wolff.models.model import Model
+from wolff.solver import Solver
+from wolff.grid import Grid
+from wolff.models.curvature_based_adaptation_model \
+    import CurvatureBasedAdaptationModel
+from wolff.sinks.grid_sink import GridSink
+from wolff.optimization.yang_cost_function import YangCostFunction
+from wolff.reinitializers.peng_reinitialization import PengReinitialization
+from wolff.optimization.finite_difference_gradient \
+    import FiniteDifferenceGradient
+import numpy as np
+
+
+class CurvatureFiniteDifferenceGradientModel(Model):
+
+    def __init__(self,
+                 dtheta=[0.1, 0.1],
+                 theta=[0.0, 0.0],
+                 grids_true=[],
+                 elapsed_times_true=[],
+                 reinitialization_iters=1):
+        super(CurvatureFiniteDifferenceGradientModel,
+              self).__init__()
+
+        self.dtheta = dtheta
+        self.theta = theta
+        self.grids_true = grids_true
+        self.elapsed_times_true = elapsed_times_true
+        self.reinitialization_iters = reinitialization_iters
+
+    @property
+    def dtheta(self):
+        return self._dtheta
+
+    @dtheta.setter
+    def dtheta(self, value):
+        if any(n < 0 for n in value):
+            raise ValueError("Cannot set simulation time to a negative number")
+        self._dtheta = list(value)
+
+    @property
+    def theta(self):
+        return self._theta
+
+    @theta.setter
+    def theta(self, value):
+        self._theta = list(value)
+
+    @property
+    def grids_true(self):
+        return self._grids_true
+
+    @grids_true.setter
+    def grids_true(self, value):
+        self._grids_true = value
+
+    @property
+    def elapsed_times_true(self):
+        return self._elapsed_times_true
+
+    @elapsed_times_true.setter
+    def elapsed_times_true(self, value):
+        self._elapsed_times_true = value
```

Comment:
Same as before about copying lists?
```suggestion
        self._elapsed_times_true = list(value)
```